File: app9/fetcher/mock_api.js
const faker = require('faker')

const getUsers = (index) => {
  faker.seed(index)

  let people = []

  for (let i = 0; i < 100; i++) {
    people.push({ uuid: faker.random.uuid(), fn: faker.name.firstName(), ln: faker.name.lastName(), friends: new Set() })
  }

  for (person of people) {
    for (friend of people) {
      if (faker.random.number() % 50 <= 2) {
        person.friends.add(friend.uuid)
        friend.friends.add(person.uuid)
      }
    }
  }

  return people.map(p => ({ ...p, friends: Array.from(p.friends)}))
}


module.exports = {
  getUsers
}

File: app11/python/mock_api.py
import random
from faker import Faker


def get_users(index: int) -> list[dict]:
    faker = Faker()
    Faker.seed(index)
    rng = random.Random(index)

    people: list[dict] = []
    for _ in range(100):
        people.append(
            {
                "uuid": faker.uuid4(),
                "fn": faker.first_name(),
                "ln": faker.last_name(),
                "friends": set(),
            }
        )

    for person in people:
        for friend in people:
            if rng.randint(0, 2**31 - 1) % 50 <= 2:
                person["friends"].add(friend["uuid"])
                friend["friends"].add(person["uuid"])

    return [
        {
            "uuid": p["uuid"],
            "fn": p["fn"],
            "ln": p["ln"],
            "friends": list(p["friends"]),
        }
        for p in people
    ]
